### PR TITLE
MOTOR-1218: Upgrade to require aiohttp >=3.8 and avoid 3.8.6

### PR DESCRIPTION
**Pull request by Jibola**

### CODE REVIEW

**Comment by Jibola** [comment]
linkcheck failures are a known issue, tracked: https://jira.mongodb.org/browse/MOTOR-1239
synchro test failures are a known issue, tracked: https://jira.mongodb.org/browse/MOTOR-1240

**Comment by blink1073** [comment]
mypy failure is unrelated (caching error on GHA).

**Review by blink1073 — APPROVED** [review]
LGTM!

**Comment by blink1073** [comment]
Actually, can you please update the title to say that we're avoiding the one version?

**Review by blink1073 — APPROVED** [review]

File: test/test_environment.py [hunk 1]
```diff
@@ -50,7 +51,12 @@
 
 HAVE_AIOHTTP = True
 try:
-    import aiohttp
+    # Remove when we drop support for 3.7
+    if TYPE_CHECKING:
+        HAVE_AIOHTTP = False
+        aiohttp = None
+    else:
+        import aiohttp
```
Comment (ShaneHarvey) [inline test/test_environment.py:59]: Can we instead add a `type ignore` comment when importing aiohttp?

File: pyproject.toml [hunk 2]
```diff
@@ -66,7 +66,7 @@ srv = [
     "pymongo[srv]>=4.5,<5",
 ]
 test = [
-    "pytest>=7", "mockupdb", "tornado>=5", "aiohttp<3.8.6", "motor[encryption]"
+    "pytest>=7", "mockupdb", "tornado>=5", "aiohttp>=3.8,!=3.8.6", "motor[encryption]"
```
Comment (ShaneHarvey) [inline pyproject.toml:69]: Why did we upgrade to `aiohttp>=3.8`?

Comment (ShaneHarvey) [reply pyproject.toml:69]: I now realize this is only the for the test suite, not the motor deps for users. So this is fine.

**Pull request merged**